lib/dispatch-rider/scheduled_job.rb

```diff
@@ -0,0 +1,59 @@
+require "active_record"
+
+# @note: Later this could be pulled out to its own gem and included depending on what ORM the user
+# prefers. Something like:
+# @example
+#  gem "dispatch_rider-active_record"
+#  gem "dispatch_rider-rom" # Ruby Object Mapper
+#  gem "dispatch_rider-mongo_mapper"
+#  gem "dispatch_rider-ohm"
+module DispatchRider
+  class ScheduledJob < ActiveRecord::Base
+    class << self
+      def publisher
+        @publisher ||= Publisher.new
+      end
+
+      # @param [ActiveSupport::Duration] every
+      # @example: DispatchRider::ScheduledJob.publish_due_jobs every 1.minute
+      def publish_due_jobs(every: nil)
+        loop {
+          claim_stub = get_new_claim_stub
+          due.unclaimed.update_all claim_stub
+          due.claimed_by(claim_stub[:claim_id]).find_each(&:publish)
+          every ? sleep(every) : break # until the next loop
+        }
```

Comment:
This loop needs to get a claim stub first.
The claim stub is valid for half an hour.
If by then the job is still not taken away then it's considered unclaimed and will be available for other loops to claim.
The loop can only claim jobs that his claim stub is valid for. No other loop could claim what this loop claimed.
The loop then published his claimed due jobs.